packages/service/src/shims/workspace.ts

```diff
@@ -274,4 +282,39 @@ export class WorkspaceShimService extends Disposable {
   async requestWorkspaceTrust() {
     return true;
   }
+  async waitForGetErr(): Promise<undefined> {
+    // Case 1: We've changed the document but haven't queued the getErr request yet. 
+    // We can tell this is the case because this.sentSeq === this.recvSeq
+    // We should wait for the request to be started.
+    // Case 2: There is an in-progress getErr request, so we can move onto the next block 
```

Comment:
This case is the `else` I guess? Or is it that there are two cases we need to wait?

I think this is confusing partly because there's an implicit case unmentioned here, which is "we are up to date"